lib/fakturoid/oauth/flow/authorization_code.rb

```diff
@@ -0,0 +1,53 @@
+# frozen_string_literal: true
+
+module Fakturoid
+  class Oauth
+    module Flow
+      class AuthorizationCode
+        include Base
+
+        GRANT_TYPE = "authorization_code"
+
+        def authorization_uri(state: nil)
+          client.config.authorization_uri(state: state)
+        end
+
+        def authorize(code:)
+          payload = {
+            grant_type: GRANT_TYPE,
+            redirect_uri: client.config.redirect_uri,
+            code: code
+          }
+
+          response = perform_request(HTTP_POST, "token.json", payload: payload)
+          client.config.credentials.update(response.body)
+          client.call_access_token_refresh_callback
```

Comment:
So you think the user should handle the update on their end manually in this case? I though this is a neat thing so you set it up only at one place.

Reply:
After more thinking about this callback, I believe it can be useful to users as you said. Maybe I would just change the name of callback. Do you think `access_token_obtained_callback` or `authorization_code_response_callback` would be more suitable?

Reply:
In the end went with `[call_]credentials_updated_callback`.